src/shared/sleep-config.c

```diff
@@ -271,5 +349,19 @@ int can_sleep(const char *verb) {
         if (!can_sleep_state(states) || !can_sleep_disk(modes))
                 return false;
 
+        /* We keep an optional white and black list (by product) to
+         * control if we want to explicitly support sleep operations. */
+        r = parse_sleep_products_config(verb, &blacklist, &whitelist);
+        if (r < 0)
+                return false;
+
+        if (is_product_listed(blacklist))
+                return false;
+
+        /* We don't support sleep operations on desktop computers
+           unless the product is explcitly white listed. */
+        if (!access("/sys/class/power_supply/BAT0", F_OK) == 0 && !is_product_listed(whitelist))
```

Comment:
I meant two separate points:
- if you want to check "am I a laptop?" you should use the hostnamed chassis property. A laptop can be without battery as well.
- as discussed on the issue, I'm perfectly fine with a blacklist-only approach, so I'm fine with removing this check entirely.